scripts/update_fixture.sh

```
@@ -32,40 +65,125 @@ function download_and_process() {
+# `latest` entry of index.json points at).
+function stub_version() {
+    jq -r '.releases[-1].version' "$WORKDIR/channel-latest.json"
+}
+
+# Under `-i`, rewrite the E2E_TEST_MOCK_INSTALL_VERSION const in tests/constant.rs
+# to the derived stub version so the atomic e2e tests target a release that
+# exists in the committed fixtures. Only done on install; a dry run must not
+# mutate a repo source file.
+function sync_test_constant() {
+    local ver="$1"
+    local const_file="$(dirname "$0")/../tests/constant.rs"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)